x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysTables.java

```diff
@@ -77,8 +76,11 @@ public final void execute(SqlSession session, ActionListener<SchemaRowSet> liste
         // namely one param specified with '%', everything else empty string
         // https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqltables-function?view=ssdt-18vs2017#comments
 
-        if (clusterPattern != null && clusterPattern.pattern().equals(SQL_WILDCARD)) {
-            if ((pattern == null || pattern.pattern().isEmpty()) && CollectionUtils.isEmpty(types)) {
+        // catalog enumeration
+        if (clusterPattern == null || clusterPattern.pattern().equals(SQL_WILDCARD)) {
+            // enumerate only if pattern is "" and no types are null
```

Comment:
The comment doesn't seem to match the code?
"no types are null" means that all types are not-null, but your check is `types == null`.